File: src/main/java/io/github/cdgeass/editor/dom/element/mapper/Choose.java
package io.github.cdgeass.editor.dom.element.mapper;

import com.intellij.util.xml.DomElement;
import com.intellij.util.xml.SubTag;
import com.intellij.util.xml.SubTagList;

import java.util.List;

/**
 * @author cdgeass
 * @since 2020-06-20
 */
public interface Choose extends DomElement {

    @SubTagList("when")
    List<When> getWhens();

    @SubTag("otherwise")
    Otherwise getOtherwise();
}

File: src/main/kotlin/io/github/cdgeass/codeInsight/dom/element/Choose.kt
// Generated on Fri Mar 26 00:14:00 CST 2021
// DTD/Schema  :    mybatis-3-mapper.dtd

package io.github.cdgeass.codeInsight.dom.element

import com.intellij.util.xml.DomElement

/**
 * mybatis-3-mapper.dtd:choose interface.
 * @author cdgeass
 */
interface Choose : DomElement {

    /**
     * Returns the list of when children.
     * @return the list of when children.
     */
    fun getWhens(): List<When>

    /**
     * Adds new child to the list of when children.
     * @return created child
     */
    fun addWhen(): When

    /**
     * Returns the value of the otherwise child.
     * @return the value of the otherwise child.
     */
    fun getOtherwise(): Otherwise
}
